src/spatio_temporal_voxel_grid.cpp

```diff
@@ -232,19 +237,20 @@ void SpatioTemporalVoxelGrid::TemporalClearAndGenerateCostmap(                \
       }
     }
     // if here, we can add to costmap and PC2
-    PopulateCostmapAndPointcloud(pt_index);
+    PopulateCostmapAndPointcloud(pt_index, true);
   }
 }
 
 /*****************************************************************************/
-void SpatioTemporalVoxelGrid::PopulateCostmapAndPointcloud(const \
-                                                            openvdb::Coord& pt)
+void SpatioTemporalVoxelGrid::PopulateCostmapAndPointcloud(                   \
```

Comment:
I think this would make more semantic sense if:

- under `if (cell != _cost_map->end())` we published if `_pub_voxels && pub_every_voxel` since its not a new insert, so only publish if meant to publish and that every voxel param is true.
- then in the `else` you can change to as `_pub_voxels`, no need to even check the param because its a pure insert either way.

Reply:
Fixed